src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,266 @@
+      }
+      
+      elToSelect.setAttribute('tabindex', 0);
+    });
+  }
+
+  #clickListener = (e) => {
+    this.handleClick(e);
+  }
+
+  // NOTE: There are definitely some "false positive" cases with multi-key pressing,
+  // but this should be good enough for most use cases.
+  #keyupListener = (e) => {
+    const { key } = e;
+    // only cancel on Escape
+    if (key === 'Escape') {
+      this.removeEventListener('keyup', this.#keyupListener);
+      return;
+    }
+
+    if (key === 'Enter' || key === ' ') {
+      this.handleSelection(e);
+    } else {
+      this.handleMovement(e);
+    }
+  }
+
+  #keydownListener = (e) => {
+    const { metaKey, altKey, key } = e;
+    if (metaKey || altKey) {
+      this.removeEventListener('keyup', this.#keyupListener);
+      return;
+    }
+    this.addEventListener('keyup', this.#keyupListener, {once: true});
+  }
+
+  enable() {
+    this.addEventListener('click', this.#clickListener);
+    this.addEventListener('keydown', this.#keydownListener);
+    this.setAttribute('tabindex', 0);
+  }
+
+  disable() {
+    this.removeEventListener('click', this.#clickListener);
+    this.removeEventListener('keyup', this.#keyupListener);
+    this.removeAttribute('tabindex');
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (attrName === MediaUIAttributes.MEDIA_CONTROLLER) {
+      if (oldValue) {
+        const mediaControllerEl = document.getElementById(oldValue);
+        mediaControllerEl?.unassociateElement?.(this);
+      }
+      if (newValue) {
+        const mediaControllerEl = document.getElementById(newValue);
+        mediaControllerEl?.associateElement?.(this);
+      }
+    } else if (attrName === 'disabled' && newValue !== oldValue) {
+      if (newValue == null) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+    }
+  }
+
+  connectedCallback() {
+    if (!this.hasAttribute('disabled')) {
+      this.enable();
+    }
+
+    this.setAttribute('role', 'listbox');
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.associateElement?.(this);
+    }
+  }
+
+  disconnectedCallback() {
+    this.disable();
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.unassociateElement?.(this);
+    }
+  }
+
+  get keysUsed() {
+    return ['Enter', ' ', 'ArrowDown', 'ArrowUp', 'Home', 'End'];
+  }
+
+  #getItem(e) {
+    if (e.target === this) return;
+
+    const composedPath = e.composedPath();
+    const slotIndex = composedPath.findIndex(el => el.nodeName === 'SLOT');
+
+    return composedPath[slotIndex - 1];
+  }
+
+  handleSelection(e) {
+    const item = this.#getItem(e);
+
```

Comment:
nit (non-blocking): is the empty newline not too spacey?

Reply:
I think better to have extra whitespace than not enough :)
But I'll keep it in mind as this gets closer to completion.